packages/next/src/server/lib/route-resolver.ts

```diff
@@ -250,15 +250,28 @@ export async function makeResolver(
 
     await router.execute(req, res, parsedUrl)
 
-    if (!res.originalResponse.headersSent) {
-      res.setHeader('x-nextjs-route-result', '1')
-      const routeResult: RouteResult = routeResults.get(req) ?? {
-        type: 'none',
-      }
+    // If the headers are sent, then this was handled by middleware and there's
+    // noting for us to do.
+    if (res.originalResponse.headersSent) {
+      return
```

Comment:
Should this call `routeResults.delete(req)`?

Reply:
If it gets to this point, then middleware would have fully handled the response before the `catchAllRoute` handler, so there shouldn't be an entry in `routeResults`